Unit3FinalAssessment/Unit3FinalAssessment/APIManager.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  APIManager.swift
+//  Unit3FinalAssessment
+//
+//  Created by Eric Chang on 11/10/16.
+//  Copyright © 2016 Eric Chang. All rights reserved.
+//
+
+import Foundation
+
+internal class APIRequestManager {
+    
+    internal static let manager: APIRequestManager = APIRequestManager()
+    private init() {}
+    
+    
+    func getData(endPoint: String, callback: @escaping (Data?) -> Void) {
+        guard let myURL = URL(string: endPoint) else { return }
+        let session = URLSession(configuration: URLSessionConfiguration.default)
+        session.dataTask(with: myURL) { (data: Data?, response: URLResponse?, error: Error?) in
+            if error != nil {
+                print("Error durring session: \(error)")
+            }
+            guard let validData = data else { return }
+            callback(validData)
+            }.resume()
+    }
+    
+    
+    
+    func downloadImage(urlString: String, callback: @escaping (Data) -> () ) {
```

Comment:
downloadImage is a redundant method compared to getData. They have the same signature and do the same thing, just some names are changed. You can use getData to get images too.